src/main/java/com/conveyal/gtfs/loader/PatternReconciliation.java

```diff
@@ -0,0 +1,455 @@
+                from = firstDifferentIndex;
+                to = lastDifferentIndex;
+                verifyInteriorStopsAreUnchanged(originalPatternIds, newReferenceIds, firstDifferentIndex, lastDifferentIndex);
+                conditionalUpdate = String.format("update %s.stop_times set stop_sequence = case " +
+                        // if sequence = fromIndex, update to toIndex.
+                        "when stop_sequence = %d then %d " +
+                        // if sequence is greater than fromIndex and less than or equal to toIndex, decrement
+                        "when stop_sequence > %d AND stop_sequence <= %d then stop_sequence - 1 " +
+                        // Otherwise, sequence remains untouched
+                        "else stop_sequence " +
+                        "end " +
+                        "from %s.trips where %s",
+                    tablePrefix, from, to, from, to, tablePrefix, joinToTrips);
+            } else if (patterns.get(firstDifferentIndex).referenceId.equals(originalPatternIds.get(lastDifferentIndex))) {
+                // Stop was moved from end of changed region to beginning of changed region (<--)
+                from = lastDifferentIndex;
+                to = firstDifferentIndex;
+                verifyInteriorStopsAreUnchanged(originalPatternIds, newReferenceIds, firstDifferentIndex, lastDifferentIndex);
+                conditionalUpdate = String.format("update %s.stop_times set stop_sequence = case " +
+                        // if sequence = fromIndex, update to toIndex.
+                        "when stop_sequence = %d then %d " +
+                        // if sequence is less than fromIndex and greater than or equal to toIndex, increment
+                        "when stop_sequence < %d AND stop_sequence >= %d then stop_sequence + 1 " +
+                        // Otherwise, sequence remains untouched
+                        "else stop_sequence " +
+                        "end " +
+                        "from %s.trips where %s",
+                    tablePrefix, from, to, from, to, tablePrefix, joinToTrips);
+            } else {
+                throw new IllegalStateException("not a simple, single move!");
+            }
+
+            // Update the stop sequences for the stop that was moved and the other stops within the changed region.
+            PreparedStatement updateStatement = connection.prepareStatement(conditionalUpdate);
+            LOG.info(updateStatement.toString());
+            int updated = updateStatement.executeUpdate();
+            LOG.info("Updated {} stop_times.", updated);
+        }
+        // CHECK IF SET OF STOPS ADDED TO END OF ORIGINAL LIST
+        else if (originalPatternIds.size() < patterns.size()) {
+            // find the left bound of the changed region to check that no stops have changed in between
+            int firstDifferentIndex = 0;
+            while (
+                firstDifferentIndex < originalPatternIds.size() &&
```

Comment:
Do we need this condition? (There is a related check on Line 271).

Reply:
Yes, so the while loop is exited without hitting an index out of bounds exception.